tests/acceptance/01_vars/02_functions/string_mustache.cf

```diff
@@ -0,0 +1,46 @@
+body common control
+{
+      inputs => { "../../default.cf.sub" };
+      bundlesequence => { default("$(this.promise_filename)") };
+      version => "1.0";
+}
+
+bundle agent init
+{
+}
+
+bundle agent test
+{
+  vars:
+      "state1" data => datastate();
+      "out11" string => string_mustache("sys.os = {{vars.sys.os}}", state1);
+      "out12" string => string_mustache("sys.os = {{vars.sys.os}}");
+
+      "state2" data => parsejson('{"x": 1, "y": [{"item": 2}, {"item": 3}, {"item": 4}]}');
+      "out21" string => string_mustache("sys.os = {{vars.sys.os}}", state2);
+      "out22" string => string_mustache("x = {{x}}", state2);
+      "out23" string => string_mustache("{{#y}}{{item}} {{/y}}", state2);
+}
+
+bundle agent check
+{
+  vars:
+      "actual" string => "
+out11 = $(test.out11)
+out12 = $(test.out12)
+out21 = $(test.out21)
+out22 = $(test.out22)
+out23 = $(test.out23)
+";
+      "expected" string => "
+out11 = sys.os = linux
+out12 = sys.os = linux
```

Comment:
Heh, I guess it should have been no surprise this failed on non-Linux platforms. You can just replace "linux" with "$(sys.os)" though, we're testing the mustache function here, not the sys variable.